sdk/iOS/src/MSUser.h

```diff
@@ -30,7 +30,7 @@
 
 
 /// The user id of the end user.
-@property (nonatomic, copy, readonly)   NSString *userId;
+@property (nonatomic, copy, readonly, nonnull)   NSString *userId;
```

Comment:
I though logout just set currentUser to nil in [MSClient](https://github.com/Azure/azure-mobile-services/blob/MobileApp/sdk/iOS/src/MSClient.m#L213-L216)